### PR TITLE
Grouping rules constructors from apexgrammar class into sub classes

### PR DESCRIPTION
**Pull request by kevinTveizaga**
refactoring apexgrammar class not code should be cleaner in that class

### CODE REVIEW

File: apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/ApexGrammar.java [hunk 1]
```diff
@@ -26,134 +26,14 @@
 import com.sonar.sslr.api.Grammar;
 import org.sonar.sslr.grammar.LexerfulGrammarBuilder;
 
+import org.fundacionjala.enforce.sonarqube.apex.api.grammar.rbuilders.DeclarationsRulesBuilder;
```
Comment (JimmyRojas) [inline apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/ApexGrammar.java]: namespace "rbuilders" ?

Comment (kevinTveizaga) [inline apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/ApexGrammar.java]: rbuilder if the package we have created to contain all grouping classes for constructors

Comment (JimmyRojas) [inline apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/ApexGrammar.java]: What does 'r' mean?

rule?

File: apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Declaration.java [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TYPE;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TYPE_CLASS;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TYPE_DECLARATION;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.VARIABLE_DECLARATION;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.VARIABLE_DECLARATOR;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.VARIABLE_DECLARATOR_ID;
+
+/**
+ *
+ * @author kevin_titichoca
```
Comment (JimmyRojas) [inline apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Declaration.java]: please remove @autor and add a brief class description instead

File: apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Expression.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.NUMERIC_EXPRESSION_OPERATIONS;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.NUMERIC_EXPRESSION_OPERATIONS_SIMPLE;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TERMINAL_EXPRESSION;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TESTING_EXPRESSION;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TYPE;
+import org.sonar.sslr.grammar.LexerfulGrammarBuilder;
+
+/**
+ *
+ * @author kevin_titichoca
```
Comment (JimmyRojas) [inline apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Expression.java]: please remove @autor and add a brief class description instead

File: apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Keyword.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import static org.fundacionjala.enforce.sonarqube.apex.api.ApexKeyword.SHARING;
+import static org.fundacionjala.enforce.sonarqube.apex.api.ApexKeyword.WITH;
+import static org.fundacionjala.enforce.sonarqube.apex.api.ApexKeyword.WITHOUT;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.KEYWORD;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.MODIFIER;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.MODIFIER_KEYWORD;
+
+/**
+ *
+ * @author kevin_titichoca
```
Comment (JimmyRojas) [inline apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Keyword.java]: please remove @autor and add a brief class description instead

File: apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Statement.java [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TERMINAL_STATEMENT;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TRY_STATEMENT;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TYPE;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.VARIABLE_DECLARATION;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.VARIABLE_DECLARATOR_ID;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.WHILE_STATEMENT;
+
+/**
+ *
+ * @author kevin_titichoca
```
Comment (JimmyRojas) [inline apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Statement.java]: please remove @autor and add a brief class description instead

File: apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Type.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+import static org.fundacionjala.enforce.sonarqube.apex.api.ApexPunctuator.AT;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.ANNOTATION;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.CLASS_NAME;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.MODIFIER;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TYPE;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.TYPE_SPECIFIER;
+
+/**
+ *
+ * @author kevin_titichoca
```
Comment (JimmyRojas) [inline apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/grammar/buildersource/Type.java]: please remove @autor and add a brief class description instead

**Comment by JimmyRojas** [comment]
Looks good to me.., thank you..

**Pull request merged**